server/controllers/profile.js

```diff
@@ -0,0 +1,90 @@
+const mongoose = require('mongoose');
+const asyncHandler = require('express-async-handler');
+
+const Profile = require('../models/Profile');
+
+// @route GET /profile
+//Search for all profiles
+exports.searchProfiles = asyncHandler(async (req,res,next) => {
+    try {
+        const profiles = await Profile.find();
+        res.status(200).json({
+          success: {
+            profiles
+          }
+        });
+      } catch (error) {
+        res.status(400);
+        throw new Error(error.message);
+      }
+})
+
+// @route POST /profile
+//Create a new profile
+exports.createProfile = asyncHandler(async (req,res,next) => {
+    const profile = req.body;
+    const newProfile = new Profile({
+        ...profile
+    });
+
+    try {
+        await newProfile.save();
+        res.status(201).json({
+          success: {
+            profile: newProfile
+          }
+        });
+        
+    } catch (error) {
+        res.status(409);
```

Comment:
What is the reasoning for using a 409 Conflict error here rather than a more general 500 Internal error?

Reply:
Hi Dan, I used the above codes as I have not had much familiarity with status codes and I picked them up from forums when I first got started with Express.

Thanks for pointing these out, I spent this morning reviewing below resources:
https://www.youtube.com/watch?v=wJa5CTIFj7U
https://www.restapitutorial.com/httpstatuscodes.html

Consequently, I have changed the status codes for the profile controllers, please review and let me know. Thanks!